### PR TITLE
Feature/mip 312

### PR DESCRIPTION
**Pull request by kkingavio**
LoggerCategoryExistsRule - Specifically checking that a category attribute exists and is not empty
LoggerMessageExistsRule - Specifically checking the message attribute

LoggerAttributesRule - Can pass a list of attributes to require

### CODE REVIEW

File: src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        GPathResult[] comps = configXml.depthFirst().findAll {
+            it.name() == componentType && it.namespaceURI() == namespace
+        }
+        comps.each { comp ->
+            Map<String, String> atts = [:]
+            comp[0].attributes.each {
+                atts.put(it.key, it.value)
+            }
+            componentList.add(new MuleComponent(atts))
+            //TODO this doesn't account for child components...
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy:36]: Can you make sure to add a story to the backlog for something like this?  This is kind of a major feature we need to have

File: src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    Boolean hasAttributeValue(String name) {
+        return attributes.get(name)?.length() > 0
+    }
+
+    String getName() {
+        if (name == null || name.length() < 1) {
+            name = attributes.get('name')
+        }
+        return name
+    }
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy:27]: We should probably differentiate between the name of the component and an attribute called name.  We should have the XML element name and namespace as part of the MuleComponent for sure, then any attribute called name would be different.

File: src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    void setName(String name) {
+        this.name = name
+    }
+
+    Integer getLineNumber() {
+        return Integer.parseInt(attributes.get('{http://www.avioconsulting.com/mule/linter}_startLineNo'))
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy:34]: We should reference the constants defined in MuleXmlParser, not hardcode the attribute namespace/name.

File: src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        }
+    }
+
+    List<LoggerComponent> findLoggerComponents() {
+        List<LoggerComponent> loggerComponents = []
+        GPathResult[] loggers = configXml.depthFirst().findAll {
+            it.name() == 'logger' && it.namespaceURI() == 'http://www.mulesoft.org/schema/mule/core'
```
Comment (adesjardin) [inline src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy:43]: Shouldn't this use findComponents?  We don't want to repeat logic for each type of component if the only difference is the name/namespace in the logic

File: src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy (hunk 1 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy:36]: https://avioconsulting.atlassian.net/browse/MIP-383

File: src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy (hunk 2 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/MuleComponent.groovy:27]: Spoke with Nisarg, he's going to make those changes in his branch.  His code relies on MuleComponent more...

File: src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy (hunk 4 above)
Comment (kkingavio) [reply src/main/groovy/com/avioconsulting/mule/linter/model/ConfigurationFile.groovy]: Made more generic.... componentName and componentNamespace are coming in Nisarg's branch, didn't want to make too many clashing changes here...

**Pull request merged**